mock/py/mockbuild/plugins/lvm_root.py

```diff
@@ -382,6 +384,22 @@ def hook_remove_snapshot(self, name):
         lvm_do(['lvremove', '-f', self.vg_name + '/' + lv_name])
         self.buildroot.root_log.info("deleted {name} snapshot".format(name=name))
 
+    def hook_recreate_snapshot(self):
+        name = self.get_current_snapshot()
```

Comment:
Why "recreate" mechanism, instead of creating a completely new snapshot?

Reply:
That's basically what I do. However, this is called only when there already is an existing snapshot of the same name (so I can't just call `make_snapshot`), which is removed and a new one is created. So it made sense to me to create a new hook for this.

Reply:
Another hook name doesn't sound entirely wrong to me, but I'm surprised that nothing
within the tho hook methods creating the snapshot from (re-)initialized chroot  can be
shared.
Also, I'd prefer a fix that would generically fix mock, not only lvm_chroot (root_cache
suffers from the same problem).

Reply:
Thinking once more about this; What if we invented `hook_postupdate` call,
and we called this hook always when some package was added during `dnf
udpate`?

Both lvm and root-cache (and maybe overlayfs) could leverage that hook
(snapshot in name doesn't match root-cache use case).

From LVM plugin, I don't know.  Do you think that we could have another
snapshot named like `update-<timestamp>` and set it as current, so users
can rollback to the `postinit` if they wanted?  (sounds better to me than just
replacing the `postinit` snapshot).

Reply:
I think it makes sense to create a new, updated snapshot. I've changed it per your suggestion. The root_cache, however, needs some more work.

Reply:
Thanks.

> The root_cache, however, needs some more work.

Isn't it just about calling `self._rebuild_root_cache()`?

Reply:
Well, `_rebuild_root_cache()` only rebuilds the cache if `self.config['cache_alterations']` is set. If that's what we want, then yes.

Reply:
Well, we can slightly alter `_rebuild_root_cache`, too (so it does something in other occasions, i.e. chroot is updated).

Reply:
I've made a change to `_rebuild_root_cache` so now it should be rebuilt in the `postupdate` hook.

Reply:
I like the current variant.  Can you please wrap the logic related to snapshot rename to separate, easily testable function?  Something like `new_name = update_snapshot_name(old_name)` (pick whatever name and input/output format, the point is to make it testable)?  Then you could add several trivial unit-tests for the function.

Could you please also enhance the lvm integration test-case so we have this functionality tested?